platform-integration/src/main/java/com/softicar/platform/integration/database/structure/DatabaseStructureJsonFromClasspathExtractor.java

```diff
@@ -0,0 +1,67 @@
+package com.softicar.platform.integration.database.structure;
+
+import com.softicar.platform.common.code.classpath.iterable.ClasspathFileIterable;
+import com.softicar.platform.common.code.classpath.metadata.ClasspathFilesMetadata;
+import com.softicar.platform.db.runtime.structure.DbRuntimeTableStructure;
+import com.softicar.platform.db.runtime.table.IDbTable;
+import com.softicar.platform.db.runtime.table.finder.DbTableFinder;
+import com.softicar.platform.db.runtime.table.row.IDbTableRow;
+import com.softicar.platform.db.structure.utils.DbTableStructureSqlGenerator;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class DatabaseStructureJsonFromClasspathExtractor {
+
+	private final String tablePackagePrefix;
+
+	public DatabaseStructureJsonFromClasspathExtractor(String tablePackagePrefix) {
+
+		this.tablePackagePrefix = Objects.requireNonNull(tablePackagePrefix);
+	}
+
+	public String extractJson() {
+
+		List<DatabaseStructureTableDefinition> definitions = loadDefinitions();
+		return new DatabaseStructureTableDefinitionsConverter().convertToJson(definitions);
+	}
+
+	private List<DatabaseStructureTableDefinition> loadDefinitions() {
+
+		return loadUniqueTables()//
+			.stream()
+			.map(this::createDefinition)
+			.collect(Collectors.toList());
+	}
+
+	private DatabaseStructureTableDefinition createDefinition(IDbTable<?, ?> table) {
+
+		return new DatabaseStructureTableDefinition(//
+			table.getFullName().getQuoted(),
+			determineCreateStatement(table));
+	}
+
+	private String determineCreateStatement(IDbTable<?, ?> table) {
+
+		var structure = new DbRuntimeTableStructure(table);
+		var sqlGenerator = new DbTableStructureSqlGenerator(structure);
+		return sqlGenerator.getCreateTableStatement();
+	}
+
+	private Collection<IDbTable<?, ?>> loadUniqueTables() {
```

Comment:
how do we ensure that the integration project depends on **all** other project, to find **all** tables?

Reply:
So far, we don't. From the top of my mind, could determine a list of all sub-projects in main `build.gradle` script, and somehow compare that to the project configuration of the `platform-integration` sub-project. Can you see an elegant way to do this (or something equivalent)?

Reply:
nope, but we might run into problems, just saying